components/wifi/esp8266-driver/ESP8266Interface.h

```diff
@@ -137,6 +137,8 @@ class ESP8266Interface : public NetworkStack, public WiFiInterface {
     /** Get the internally stored IP address
      *  @return             IP address of the interface or null if not yet connected
      */
+    virtual nsapi_error_t get_ip_address(SocketAddress *address);
+
     virtual const char *get_ip_address();
```

Comment:
Should we mark these as deprecated as well?

Reply:
I pushed an update deprecating string-based API of `ESP8266` driver. I think this is a matter of being consequent. We cannot forbid member function overloading, but at least we can give a good example.

Another thing is that users will most likely use `NetworkInterface::get_default_instance()` rather that consciously creating `EPS8266Interface`, so they should be given deprecation warnings if they use string-based APIs anyway?